In [1]:
import pandas as pd


In [2]:
#%pip install openpyxl

There are different file formats in the 2020 files so I am going to read each one in by itself with for loop and just pull the Villagers and Tally columns since that is all we need and then combine the 2021 and 2020 data sets into one total dataframe to link to the Villagers.csv and attach the ACP_Poll_Votes aka "Tally".  We might need to clean some names.  

In [3]:
#just trying out and seeing/checking that the villagers and tally columns are there
acpJune20 = pd.read_excel('PollResults/2020/ACP - June 2020 Poll Final Vote Count.xlsx', sheet_name = 'Sheet1')

In [4]:
acpJune20.head()

Timestamp  1st Vote  2nd Vote 3rd Vote  Unnamed: 4 Villagers  \
0 2020-06-01 00:03:19.000     Poppy   Raymond   Bunnie         NaN   Marshal   
1 2020-06-01 00:08:45.000      Wade       Dom     Gala         NaN   Raymond   
2 2020-06-01 00:09:26.000  Wolfgang     Dobie    Drago         NaN     Sherb   
3 2020-06-01 00:10:21.655    Bianca  Freckles    Pango         NaN    Zucker   
4 2020-06-01 00:10:58.205     Kitty     Pecan    Poppy         NaN      Judy   

    Tally  Unnamed: 7                                            Comment  
0  6044.0         NaN  If a villager has the same votes, they're rank...  
1  5656.0         NaN                                                NaN  
2  3374.0         NaN                                                NaN  
3  3079.0         NaN                                                NaN  
4  2696.0         NaN                                                NaN

In [5]:
import os
ACP_PollRes2020 = pd.DataFrame()
for root, dirs, files in os.walk('PollResults/2020/'):
    for file in files:
        filename = file
        print(filename)
        if 'ACP' in filename:
            df = pd.read_excel('PollResults/2020/'+filename)
            df = df[['Villagers','Tally']]
            #df = df[df['Villagers']].notna()
            ACP_PollRes2020 = ACP_PollRes2020.append(df)

ACP - December Poll Final Vote Count 2020.xlsx
ACP - August 2020 Poll Final Vote Count.xlsx
ACP - September 2020 Poll Final Vote Count.xlsx
ACP - November 2020 Poll Final Vote Count.xlsx
ACP - June 2020 Poll Final Vote Count.xlsx
ACP - October 2020 Poll Final Vote Count.xlsx
ACP - July 2020 Poll Final Vote Count.xlsx


In [6]:
ACP_PollRes2020.shape

(150743, 2)

In [7]:
ACP_PollRes2020.head()

Villagers  Tally
0   Marshal  936.0
1   Raymond  872.0
2     Sherb  651.0
3    Zucker  602.0
4      Judy  472.0

In [8]:
ACP_PollRes2020 = ACP_PollRes2020[ACP_PollRes2020['Villagers'].notna()]

In [9]:
ACP_PollRes2020.shape

(2737, 2)

In [10]:
ACP_PollRes2020.Villagers.unique()

array(['Marshal', 'Raymond', 'Sherb', 'Zucker', 'Judy', 'Marina', 'Erik',
       'Merry', 'Fauna', 'Stitches', 'Ankha', 'Beau', 'Lolly', 'Audie',
       'Dom', 'Merengue', 'Maple', 'Bob', 'Coco', 'Diana', 'Molly',
       'Lucky', 'Roald', 'Ketchup', 'Poppy', 'Cherry', 'Tangy', 'Fang',
       'Apollo', 'Rosie', 'Goldie', 'Julian', 'Whitney', 'Apple',
       'Filbert', 'Tia', 'Sprinkle', 'Pietro', 'Octavian', 'Cookie',
       'Flora', 'Bluebear', 'Punchy', 'Kiki', 'Flurry', 'Genji', 'Skye',
       'Pashmina', 'Melba', 'Pekoe', 'Aurora', 'Lily', 'Zell', 'Ruby',
       'Dobie', 'Wolfgang', 'Kid Cat', 'Chevre', 'Bam', 'Agnes', 'Clay',
       'Bones', 'Bianca', 'Henry', 'Kabuki', 'Daisy', 'Kyle', 'Shep',
       'Reneigh', 'Katt', 'Hornsby', 'Chrissy', 'Peanut', 'Muffy',
       'Sylvana', 'Phoebe', 'Bunnie', 'Fuchsia', 'Hamlet', 'Deirdre',
       'Wade', 'June', 'Antonio', 'Bill', 'Cheri', 'Gayle', 'Rudy',
       'Felicity', 'Mitzi', 'Chief', 'Lyman', 'Ozzie', 'Snake', 'Dotty',
       'Papi',

In [11]:
ACP_2020 = ACP_PollRes2020.groupby(by = 'Villagers',dropna=True).Tally.sum().reset_index()

In [12]:
ACP_2020 = ACP_2020.sort_values(['Tally'], ascending = False)

In [13]:
ACP_2020 = ACP_2020.reset_index(drop = True)

In [14]:
ACP_2020.head(10)

Villagers    Tally
0   Marshal  20735.0
1   Raymond  19190.0
2     Sherb  13217.0
3    Zucker  10956.0
4      Judy   9816.0
5    Marina   8773.0
6     Merry   8190.0
7     Audie   7896.0
8  Stitches   7870.0
9     Ankha   7832.0

Doing similar process for the 2021 poll result files from the Animal Crossing Portal player voting results

In [15]:
import os
ACP_PollRes2021 = pd.DataFrame()
for root, dirs, files in os.walk('PollResults/2021/'):
    for file in files:
        filename = file
        print(filename)
        if 'ACP' in filename:
            df = pd.read_excel('PollResults/2021/'+filename)
            df = df[['Villagers','Tally']]
            #df = df[df['Villagers']].notna()
            ACP_PollRes2021 = ACP_PollRes2021.append(df)

ACP - May 2021 Poll Final Vote Count.xlsx
ACP - February Poll Final Vote Count 2021.xlsx
ACP - August 2021 Poll Final Vote Count.xlsx
ACP - September 2021 Poll Final Vote Count.xlsx
ACP - July 2021 Poll Final Vote Count.xlsx
ACP - January Poll Final Vote Count 2021.xlsx
ACP - April 2021 Poll Final Vote Count.xlsx
ACP - October 2021 Villager Popularity Poll (Responses).xlsx
ACP - March 2021 Poll Final Vote Count.xlsx
ACP - June 2021 Poll Final Vote Count.xlsx


In [16]:
ACP_PollRes2021.shape

(41499, 2)

In [17]:
ACP_PollRes2021.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41499 entries, 0 to 2426
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Villagers  3958 non-null   object 
 1   Tally      3958 non-null   float64
dtypes: float64(1), object(1)
memory usage: 972.6+ KB


In [18]:
ACP_PollRes2021.head()

Villagers  Tally
0   Marshal  479.0
1   Raymond  459.0
2    Zucker  413.0
3     Sherb  390.0
4      Judy  323.0

In [19]:
#ACP_PollRes2021 = ACP_PollRes2021[ACP_PollRes2021['Villagers']].fillna(0)

In [20]:
ACP_PollRes2021.Villagers.unique()

array(['Marshal', 'Raymond', 'Zucker', 'Sherb', 'Judy', 'Stitches',
       'Ankha', 'Marina', 'Fauna', 'Audie', 'Molly', 'Merengue', 'Lolly',
       'Maple', 'Beau', 'Dom', 'Erik', 'Coco', 'Diana', 'Bob', 'Lucky',
       'Ketchup', 'Cherry', 'Poppy', 'Roald', 'Julian', 'Merry', 'Étoile',
       'Apple', 'Apollo', 'Rosie', 'Bluebear', 'Cookie', 'Pietro', 'Tia',
       'Tangy', 'Whitney', 'Filbert', 'Octavian', 'Melba', 'Fang', 'Lily',
       'Marty', 'Pekoe', 'Goldie', 'Punchy', 'Flora', 'Kiki', 'Kabuki',
       'Sprinkle', 'Wolfgang', 'Zell', 'Reneigh', 'Skye', 'Agnes', 'Chai',
       'Kid Cat', 'Genji', 'Chrissy', 'Aurora', 'Bam', 'Daisy',
       'Pashmina', 'Bill', 'Bianca', 'Bones', 'Flurry', 'Phoebe', 'Dobie',
       'Ruby', 'Fuchsia', 'Bunnie', 'Shep', 'Ozzie', 'Hornsby', 'Chevre',
       'Gayle', 'Kyle', 'Deirdre', 'Muffy', 'Katt', 'Alfonso', 'Clay',
       'Drago', 'Peanut', 'Static', 'Mitzi', 'Julia', 'Bea', 'Papi',
       'Sylvana', 'Henry', 'Dotty', 'Hazel', 'Rudy', 'Francine

In [21]:
ACP_2021 = ACP_PollRes2021.groupby(by = 'Villagers',dropna=True).Tally.sum().reset_index()

In [22]:
ACP_2021 = ACP_2021.sort_values(['Tally'], ascending = False)

In [23]:
ACP_2021 = ACP_2021.reset_index(drop = True)

In [24]:
ACP_2021.head(10)

Villagers   Tally
0   Marshal  5865.0
1   Raymond  5375.0
2    Zucker  4386.0
3     Sherb  4278.0
4      Judy  3238.0
5  Stitches  2997.0
6     Ankha  2979.0
7    Marina  2950.0
8     Fauna  2925.0
9     Audie  2655.0

Stacking the 2020 and 2021 dataframe from ACP Results

In [25]:
# Stack the DataFrames on top of each other
ACP_FinalDF = pd.concat([ACP_2020, ACP_2021], axis=0)

In [26]:
ACP_FinalDF.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 789 entries, 0 to 396
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Villagers  789 non-null    object 
 1   Tally      789 non-null    float64
dtypes: float64(1), object(1)
memory usage: 18.5+ KB


Repeating the same groupby process about to the get the total tally sum of voted from Animal Crossing Portal by villager

In [27]:
ACP_FinalDF = ACP_FinalDF.groupby(by = 'Villagers',dropna=True).Tally.sum().reset_index()

In [28]:
ACP_FinalDF = ACP_FinalDF.sort_values(['Tally'], ascending = False)

In [29]:
ACP_FinalDF = ACP_FinalDF.reset_index(drop = True)

In [30]:
ACP_FinalDF.head(10)

Villagers    Tally
0   Marshal  26600.0
1   Raymond  24565.0
2     Sherb  17495.0
3    Zucker  15342.0
4      Judy  13054.0
5    Marina  11723.0
6  Stitches  10867.0
7     Ankha  10811.0
8     Audie  10551.0
9     Fauna  10248.0

In [31]:
ACP_FinalDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Villagers  398 non-null    object 
 1   Tally      398 non-null    float64
dtypes: float64(1), object(1)
memory usage: 6.3+ KB


I think we have some name variations of the same villagers, we will have to clean those a bit.

Goal was to check the villager names but will have to convert to list and then sort ascending and look at the names independently.  

In [32]:
chkdupenames2021 = ACP_FinalDF.groupby(by='Villagers').count()

In [33]:
chkdupenames2021.reset_index().sort_values(by = 'Tally', ascending = False)

Villagers  Tally
0     Admiral      1
273     Pecan      1
271   Peaches      1
270     Paula      1
269     Patty      1
..        ...    ...
130     Elvis      1
129    Eloise      1
128     Elmer      1
127     Ellie      1
397    Étoile      1

[398 rows x 2 columns]

In [34]:
chkdupenames2021.Tally.unique()

array([1])

So checking for dupes didn't work as I thought but still good to check for sanity

Going to take the villagers column in df and push to a list then sort ascending.

In [35]:
chknames = ACP_FinalDF.Villagers.to_list()

In [36]:
chknames = sorted(chknames)

In [37]:
chknames

['Admiral',
 'Agent S',
 'Agnes',
 'Al',
 'Alfonso',
 'Alice',
 'Alli',
 'Amelia',
 'Anabelle',
 'Anchovy',
 'Angus',
 'Anicotti',
 'Ankha',
 'Annalisa',
 'Annalise',
 'Antonio',
 'Apollo',
 'Apple',
 'Astrid',
 'Audie',
 'Aurora',
 'Ava',
 'Avery',
 'Axel',
 'Baabara',
 'Bam',
 'Bangle',
 'Barold',
 'Bea',
 'Beardo',
 'Beau',
 'Becky',
 'Bella',
 'Benedict',
 'Benjamin',
 'Bertha',
 'Bettina',
 'Bianca',
 'Biff',
 'Big Top',
 'Bill',
 'Billy',
 'Biskit',
 'Bitty',
 'Blaire',
 'Blanche',
 'Bluebear',
 'Bob',
 'Bonbon',
 'Bones',
 'Boomer',
 'Boone',
 'Boots',
 'Boris',
 'Boyd',
 'Bree',
 'Broccolo',
 'Broffina',
 'Bruce',
 'Bubbles',
 'Buck',
 'Bud',
 'Bunnie',
 'Butch',
 'Buzz',
 'Cally',
 'Camofrog',
 'Canberra',
 'Candi',
 'Carmen',
 'Caroline',
 'Carrie',
 'Cashmere',
 'Celia',
 'Cesar',
 'Chadder',
 'Chai',
 'Charlise',
 'Chelsea',
 'Cheri',
 'Cherry',
 'Chester',
 'Chevre',
 'Chief',
 'Chops',
 'Chow',
 'Chrissy',
 'Claude',
 'Claudia',
 'Clay',
 'Cleo',
 'Clyde',
 'Coach',
 'Cob

Need to check Annalise and Annalisa (edit: checked anteater and horse so okay), Curlos and Curly(edit: sheep and pig), Del and Deli(edit: alligator and monkey), Remove Nan (edit: is a goat), Tammi and Tammy (edit: monkey and cub(bear)), Ursula and Ursala (edit: need to change Ursula to Ursala, just one character/villager that is a cub(bear)).  (Note: Possibly switch O'Hare to OHare and fix Renee with tilde for joining purposes) 

In [38]:
#fixing Ursula to Ursala
ACP_FinalDF["Villagers"].replace({"Ursula": "Ursala"}, inplace=True)

In [39]:
ACP_FinalDF.head()

Villagers    Tally
0   Marshal  26600.0
1   Raymond  24565.0
2     Sherb  17495.0
3    Zucker  15342.0
4      Judy  13054.0

In [47]:
#redoing groupbyto fix Ursala totals
ACP_FinalDF = ACP_FinalDF.groupby(by = 'Villagers',dropna=True).Tally.sum().reset_index()

In [48]:
ACP_FinalDF = ACP_FinalDF.sort_values(['Tally'], ascending = False)

In [49]:
ACP_FinalDF = ACP_FinalDF.reset_index(drop = True)

In [50]:
ACP_FinalDF.head()

Villagers    Tally
0   Marshal  26600.0
1   Raymond  24565.0
2     Sherb  17495.0
3    Zucker  15342.0
4      Judy  13054.0

In [52]:
#redoing chknames
chknames = ACP_FinalDF.Villagers.to_list()

In [53]:
chknames = sorted(chknames)

In [54]:
print(len(chknames))

397


Ursula to Ursala is now fixed - we can tell by length which went down from 398 to 397

In [55]:
villagers = pd.read_csv('villagers.csv')

In [56]:
villagers.head()

Name    Species  Gender Personality    Hobby Birthday Catchphrase  \
0  Admiral       Bird    Male      Cranky   Nature   27-Jan     aye aye   
1  Agent S   Squirrel  Female       Peppy  Fitness    2-Jul    sidekick   
2    Agnes        Pig  Female  Big Sister     Play   21-Apr     snuffle   
3       Al    Gorilla    Male        Lazy  Fitness   18-Oct     ayyyeee   
4  Alfonso  Alligator    Male        Lazy     Play    9-Jun    it'sa me   

   Favorite Song Style 1  Style 2 Color 1 Color 2               Wallpaper  \
0     Steep Hill    Cool     Cool   Black    Blue          dirt-clod wall   
1  Go K.K. Rider  Active   Simple    Blue   Black           concrete wall   
2     K.K. House  Simple  Elegant    Pink   White  gray molded-panel wall   
3  Go K.K. Rider  Active   Active     Red   White           concrete wall   
4    Forest Life  Simple   Simple     Red    Blue    yellow playroom wall   

                 Flooring                                     Furniture List  \
0                  tatami  717;1849;7047;2736;787;5970;3449;3622;3802;410...   
1  colorful tile flooring  7845;7150;3468;4080;290;3971;3449;1708;4756;25...   
2      arabesque flooring  4129;7236;7235;7802;896;3428;4027;7325;3958;71...   
3   green rubber flooring  1452;4078;4013;833;4116;3697;7845;3307;3946;39...   
4    green honeycomb tile  4763;3205;3701;1557;3623;85;3208;3584;4761;121...   

  Filename    Unique Entry ID  
0    brd06  B3RyfNEqwGmcccRC3  
1    squ05  SGMdki6dzpDZyXAw5  
2    pig17  jzWCiDPm9MqtCfecP  
3    gor08  LBifxETQJGEaLhBjC  
4    crd00  REpd8KxB8p9aGBRSE

In [57]:
villagerschk = villagers['Name'].to_list()

In [58]:
print(len(villagerschk))

391


In [59]:
print(len(chknames))

397


In [60]:
setV = set(villagerschk)
setC = set(chknames)

In [61]:
missing = list(sorted(setV - setC))
added = list(sorted(setC - setV))

print('missing:', missing)
print('added:', added)

missing: []
added: ['Chai', 'Chelsea', 'Marty', 'Rilla', 'Toby', 'Étoile']


Need to check why 'Chai','Chelsea', 'Marty', 'Rilla', 'Toby','Étoile' are not in villagers.csv file are they brand new?

Upon further research these villagers: Chai, Chelsea, Marty, Rilla, Toby and Etoile are from the Sanrio update so they will not be in our Villagers datset.  When joining the two data sets villagers.csv and the ACP_FinalDF they will disappear :) 

In [62]:
ACP_FinalDF.to_csv("ACP_FinalDF.csv")